### PR TITLE
[ISSUE-3091][Improve] Improve streampark-flink module based on [3.10 Pre-Conditions Checking]

### LINKED ISSUE #3091: [Improve] Improve `streampark-flink` module based on [3.10 Pre-Conditions Checking]
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-streampark/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

[Improve] Improve `streampark-flink` module based on [3.10 Pre-Conditions Checking](https://streampark.apache.org/community/submit_guide/code_style_and_quality_guide#310-pre-conditions-checking)

### Usage Scenario

_No response_

### Related issues

#2986

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Xuxiaotuan**
## What changes were proposed in this pull request

Issue Number: close #3091 

Use a unified AssertUtils.require() to complete the validation of the prerequisite

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): no

### CODE REVIEW

File: streampark-flink/streampark-flink-sql-gateway/streampark-flink-sql-gateway-base/src/main/java/org/apache/streampark/gateway/factories/FactoryUtil.java [hunk 1]
```diff
@@ -48,47 +48,44 @@ public static <T extends Factory> T discoverFactory(
...
-                  + "%s",
-              factoryIdentifier,
-              factoryClass.getName(),
-              matchingFactories.stream()
-                  .map(f -> f.getClass().getName())
-                  .sorted()
-                  .collect(Collectors.joining("\n"))));
-    }
+    AssertUtils.required(
+        !matchingFactories.isEmpty(),
```
Comment (RocMarshal) [inline streampark-flink/streampark-flink-sql-gateway/streampark-flink-sql-gateway-base/src/main/java/org/apache/streampark/gateway/factories/FactoryUtil.java]: There's a un-matched semantic replacement.
The  original potential  Exception is `ValidationException` but the it's `IllegalArgumentxx` exception.
In other words, the line is not suitable for the change.

Would you mind checking the whole corresponding changed lines about exception-matched ?

**Review by RocMarshal — COMMENTED** [review]
Thanks @Xuxiaotuan for the contribution.
I left a few of comments. 
PTAL in your free time~

**Comment by Xuxiaotuan** [comment]
@RocMarshal  I restored the changes to the ValidationException

**Review by RocMarshal — APPROVED** [review]
Thanks @Xuxiaotuan for the quick update.
LGTM +1 now.
CC @caicancai

**Pull request merged**